src/app.ts

```diff
@@ -493,8 +493,13 @@ app.post('/groups/actions/reset', async (req, res, next) => {
 app.post(
     '/groups/:name/actions/launch-protected',
     body('count').isInt({ min: 0 }).withMessage('Value must be positive'),
+    body('maxDesired').optional().isInt({ min: 0 }).withMessage('Value must be positive'),
     body('count').custom(async (value, { req }) => {
-        if (!(await validator.canLaunchInstances(req.params.name, value))) {
+        let max = 0;
+        if (req.body.maxDesired) {
+            max = req.body.maxDesired;
+        }
+        if (!(await validator.canLaunchInstances(req.params.name, value, max))) {
```

Comment:
WDYT about sending the whole req as a parameter to canLaunchInstances, and inside canLaunchInstances, using a similar logic to launchProtectedInstanceGroup, to decide whether to override max? Something like:
if (requestBody.maxDesired != null) {
     max = requestBody.maxDesired;
} else {
    max = instanceGroup.scalingOptions.maxDesired;
}